crates/wasmparser/src/validator/operators.rs

```diff
@@ -4398,24 +4440,34 @@ where
         Ok(())
     }
     fn visit_any_convert_extern(&mut self) -> Self::Output {
-        let extern_ref = self.pop_operand(Some(RefType::EXTERNREF.into()))?;
-        let is_nullable = extern_ref
-            .as_type()
-            .map_or(false, |ty| ty.as_reference_type().unwrap().is_nullable());
+        let extern_ref = self.pop_maybe_shared_ref(RefType::EXTERNREF)?;
+        let (is_nullable, shared) = if let Some(extern_ref) = extern_ref {
+            (
+                extern_ref.is_nullable(),
+                self.resources.is_shared_ref_type(extern_ref),
+            )
+        } else {
+            (false, false)
```

Comment:
This is something that might be worth clarifying in the upstream shared-everything-threads proposal. Specifically once unreachable code is being validated wha's the value of `shared` to use here? The nullable flag works out since it's easy to pick a "top type" that everything else is a subtype, but that's not true for `shared`. In theory this is invalid because later instructions might be expecting `shared` and that should work given the high-level description of unreachable code (IIRC) but won't work in the implementation of this algorithm.

Reply:
https://github.com/WebAssembly/shared-everything-threads/issues/80

Reply:
Thanks! Mind leaving a TODO here and in `extern_convert_any` pointing to that issue?